### PR TITLE
Us17 01 profil

### PR DESCRIPTION
**Pull request by wonecode**

### CODE REVIEW

**Review by bouboumael — DISMISSED** [review]
ok just many files because autogenerate with symfony.

File: src/Entity/RegisteredUser.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    /**
+     * @ORM\OneToOne(targetEntity=Subscription::class, cascade={"persist", "remove"})
+     * @ORM\JoinColumn(nullable=true)
+     */
+    private Subscription $subscription;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private string $username;
```
Comment (LucyMerlier) [inline src/Entity/RegisteredUser.php:65]: you should add validations right away (using @ Assert annotations)

File: src/Entity/Subscription.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private string $curriculum;
+
+    /**
+     * @ORM\Column(type="text", nullable=true)
+     */
+    private string $jobDescription;
```
Comment (LucyMerlier) [inline src/Entity/Subscription.php:35]: you should add validations right away (using @ Assert annotations)

File: src/Form/RegisteredUserType.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class RegisteredUserType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('pseudo')
```
Comment (LucyMerlier) [inline src/Form/RegisteredUserType.php]: from what i understand, this property has been renamed "username"?

File: src/Form/RegisteredUserType.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            ->add('firstname')
+            ->add('lastname')
+            ->add('phone')
+            ->add('address')
+            ->add('jobAddress')
+            ->add('ogr')
+            ->add('user')
+            ->add('subscription')
```
Comment (LucyMerlier) [inline src/Form/RegisteredUserType.php]: i feel like you should have a form that creates the "User" entity associated with the "RegisteredUser" entity created here => "user" and "subscription" fields should not be here

File: src/Form/RegisteredUserType.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        ;
+    }
```
Comment (LucyMerlier) [inline src/Form/RegisteredUserType.php]: you should add options right away when you create a form (labels, etc...)

File: src/Form/RegisteredUserType.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => RegisteredUser::class,
+        ]);
+    }
+}
```
Comment (LucyMerlier) [inline src/Form/RegisteredUserType.php]: you are not using this form anywhere in your website yet

File: src/Form/SubscriptionType.php [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        ;
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => Subscription::class,
+        ]);
+    }
+}
```
Comment (LucyMerlier) [inline src/Form/SubscriptionType.php]: same comments as with other form class

File: templates/profile/show.html.twig [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+						<h2 class="h3 m-0">Développeuse PHP</h2>
+					</div>
+				</div>
+				<i class="far fa-heart"></i>
+			</div>
+			<div class="profile-body mt-5">
+				<div class="mb-5">
+					<div class="d-flex flex-direction-column align-items-center mb-4">
+						<i class="fas fa-briefcase me-4"></i>
+						<p class="m-0 p-0">Orléans, Loiret</p>
```
Comment (LucyMerlier) [inline templates/profile/show.html.twig]: isn't this specified dynamically thanks to the "jobAddress" property in "registeredUser"?

File: templates/profile/show.html.twig [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+					</div>
+					<div class="d-flex flex-direction-column align-items-center mb-4">
+						<i class="fas fa-home me-4"></i>
+						<p class="m-0 p-0">Tours, Indre-et-Loire</p>
```
Comment (LucyMerlier) [inline templates/profile/show.html.twig]: isn't this specified dynamically thanks to the "address" property in "registeredUser"?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

**Review by bouboumael — APPROVED** [review]

**Review by LucyMerlier — APPROVED** [review]

**Pull request merged**